src/web/releases.rs

```diff
@@ -1040,11 +1042,128 @@ mod tests {
     }
 
+    #[test]
+    fn recent_releases() {
+        wrapper(|env| {
+            let web = env.frontend();
+            assert_success("/releases", web)?;
+
+            env.db().fake_release().name("some_random_crate").create()?;
+            env.db()
+                .fake_release()
+                .name("some_random_crate_that_failed")
+                .build_result_successful(false)
+                .create()?;
+            assert_success("/releases", web)
+        })
+    }
+
+    #[test]
+    fn recent_releases_by_stars() {
+        wrapper(|env| {
+            let web = env.frontend();
+            assert_success("/releases/stars", web)?;
+
+            env.db().fake_release().name("some_random_crate").create()?;
+            env.db()
+                .fake_release()
+                .name("some_random_crate_that_failed")
+                .build_result_successful(false)
+                .create()?;
+            assert_success("/releases/stars", web)
+        })
+    }
+
+    #[test]
+    fn recent_failures() {
+        wrapper(|env| {
+            let web = env.frontend();
+            assert_success("/releases/recent-failures", web)?;
+
+            env.db().fake_release().name("some_random_crate").create()?;
+            env.db()
+                .fake_release()
+                .name("some_random_crate_that_failed")
+                .build_result_successful(false)
+                .create()?;
+            assert_success("/releases/recent-failures", web)
+        })
+    }
+
+    #[test]
+    fn failures_by_stars() {
+        wrapper(|env| {
+            let web = env.frontend();
+            assert_success("/releases/failures", web)?;
+
+            env.db().fake_release().name("some_random_crate").create()?;
+            env.db()
+                .fake_release()
+                .name("some_random_crate_that_failed")
+                .build_result_successful(false)
+                .create()?;
+            assert_success("/releases/failures", web)
+        })
+    }
+
+    #[test]
+    fn release_activity() {
+        wrapper(|env| {
+            let web = env.frontend();
+            assert_success("/releases/activity", web)?;
+
+            env.db().fake_release().name("some_random_crate").create()?;
+            env.db()
+                .fake_release()
+                .name("some_random_crate_that_failed")
+                .build_result_successful(false)
+                .create()?;
+            assert_success("/releases/activity", web)
+        })
+    }
+
+    #[test]
+    fn recent_queue() {
+        wrapper(|env| {
+            let web = env.frontend();
+            assert_success("/releases/queue", web)?;
+
+            env.db().fake_release().name("some_random_crate").create()?;
+            env.db()
+                .fake_release()
+                .name("some_random_crate_that_failed")
+                .build_result_successful(false)
+                .create()?;
+            assert_success("/releases/queue", web)
```

Comment:
This doesn't test what you think it does. `fake_release` does not touch the build queue at all, this is the same test as if you hadn't had the call to `fake_release()`.

Reply:
Fixed to test against a populated queue in 3fff667

Reply:
Hmm, I'd remove that test -- I added a better one in #863, and your copy uses an API that my PR removed.